apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/Tasks/TaskAwaitable.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.PubSub.V1.Tasks
+{
+    internal struct TaskAwaitable
+    {
+        private class Awaiter : ITaskAwaiter
+        {
+            public Awaiter(Task task, TaskScheduler taskScheduler)
+            {
+                _task = task;
+                _taskScheduler = taskScheduler;
+            }
+
+            private Task _task;
+            private TaskScheduler _taskScheduler;
+
+            public void OnCompleted(Action continuation) => _task.ContinueWith(_ => continuation(), _taskScheduler);
+            public void UnsafeOnCompleted(Action continuation) => _task.ContinueWith(_ => continuation(), _taskScheduler);
```

Comment:
I get nervous seeing code like this not in test code. I can't remember all the details about what OnCompleted vs UnsafeOnCompleted are meant to do, but I suspect they're important. Let's chat about this.

(I'm keen on testability, but ideally we shouldn't have any code like this in the prod package.)